### PR TITLE
Drop timelib dependency

### PR DESCRIPTION
**Pull request by prophile**
timelib appears to be unmaintained and doesn't compile on MacOS.

### CODE REVIEW

File: sr/comp/cli/add_delay.py [hunk 1]
```diff
@@ -25,10 +25,45 @@ def parse_duration(time_str):
...
+
+    def parse_past(match):
+        offset = parse_duration(match.group(1))
+        return datetime.now() - offset
+
+    def parse_absolute(match):
+        return parse_date(match.group(0))
+
+    DATETIME_PATTERNS = [
+        (r'^([^ ]+)\s*ago$', parse_past),
```
Comment (PeterJCLaw) [inline sr/comp/cli/add_delay.py:48]: This looks like it'll match `5mago` is that desired?

File: sr/comp/cli/add_delay.py [hunk 2]
```diff
@@ -25,10 +25,45 @@ def parse_duration(time_str):
...
+        (r'^in\s*([^ ]+)$', parse_future),
```
Comment (PeterJCLaw) [inline sr/comp/cli/add_delay.py:49]: This looks like it'll match `in5m` is that desired?

**Review by PeterJCLaw — APPROVED** [review]
Nice, thanks.

File: sr/comp/cli/add_delay.py (hunk 1 above)
Comment (prophile) [reply sr/comp/cli/add_delay.py:48]: That string's kind of janky but it is also unambiguous, I thought it was probably fine to let it through—thoughts?

Comment (PeterJCLaw) [reply sr/comp/cli/add_delay.py:48]: Yeah, let's :shipit:. If it turns out to be an issue we can make it more strict.

**Pull request merged**